Index: b/panels/keyboard/gnome-keyboard-panel.desktop.in.in
===================================================================
--- a/panels/keyboard/gnome-keyboard-panel.desktop.in.in
+++ b/panels/keyboard/gnome-keyboard-panel.desktop.in.in
@@ -2,7 +2,7 @@
 _Name=Keyboard
 _Comment=Change keyboard settings
 Exec=cinnamon-control-center keyboard
-Icon=input-keyboard
+Icon=preferences-desktop-keyboard
 Terminal=false
 Type=Application
 StartupNotify=true
