# Description: files to translate from ubuntu patches
Index: cinnamon-control-center-3.6.2/po/POTFILES.in
===================================================================
--- cinnamon-control-center-3.6.2.orig/po/POTFILES.in	2012-11-02 23:34:11.946702529 -0400
+++ cinnamon-control-center-3.6.2/po/POTFILES.in	2012-11-02 23:35:25.194699465 -0400
@@ -81,6 +87,7 @@
 [type: gettext/glade]panels/region/cinnamon-region-panel-input-chooser.ui
 panels/region/cinnamon-region-panel-system.c
 [type: gettext/glade]panels/region/cinnamon-region-panel.ui
+panels/region/ubuntu-region-panel.desktop.in.in
 panels/screen/cinnamon-screen-panel.desktop.in.in
 [type: gettext/glade]panels/screen/screen.ui
 panels/sound/applet-main.c
@@ -135,3 +142,4 @@
 shell/gnomecc.directory.in
 shell/cinnamon-control-center.desktop.in.in
 [type: gettext/glade]shell/shell.ui
+shell/cc-shell-nav-bar.c
