Index: gnome-control-center-3.4.2/panels/keyboard/01-launchers.xml.in
===================================================================
--- gnome-control-center-3.4.2.orig/panels/keyboard/01-launchers.xml.in	2011-08-25 11:09:31.000000000 -0400
+++ gnome-control-center-3.4.2/panels/keyboard/01-launchers.xml.in	2012-08-25 23:34:01.889092065 -0400
@@ -7,6 +7,8 @@
 
 	<KeyListEntry name="email" _description="Launch email client"/>
 
+	<KeyListEntry name="terminal" _description="Launch terminal"/>
+
 	<KeyListEntry name="www" _description="Launch web browser"/>
 
 	<KeyListEntry name="home" _description="Home folder"/>
